### PR TITLE
Reduce size

### PR DESCRIPTION
**Pull request by markusenglund**
Hey, I'm a big fan of the library.

Since reducing bundle-size is an important focus for this library, I went in and did some more optimizations. What I did boils down to this:
* Use Rollup instead of only babel for building the library. This is where the majority of the size savings come from.
* Remove prop-types when in production mode using [this plugin](https://www.npmjs.com/package/babel-plugin-transform-react-remove-prop-types). This lets the library user get the benefits of the prop-type warnings when they're in development mode, but removes them from the final bundle in production.
* Use babels "loose" setting which creates smaller, but not quite as spec-compliant output. (In my experience very safe to use)

I used an example project to measure the savings with webpack-bundle-analyzer. The measured savings were as follows:
* Minified: 22.85kb -> 20.16kb
* Minified + Gzipped: 7.13kb -> 6.44kb

So basically 10% reduction. Looking forward to any questions or feedback.

### CODE REVIEW

**Comment by mariusandra** [comment]
Hey! Thank you for this and my apologies for the delay in merging.
I'll release `0.12` soon with this and a few other changes.

**Comment by markusenglund** [comment]
Cool! No problem.

**Comment by mariusandra** [comment]
And `0.12.0` is released! Thanks again!
